File: exercises/week05/src/Strategy/Java/TextEditor.java
package Strategy.Java;


public class TextEditor {

    private TextFormatter formatter;

    public TextEditor(TextFormatter formatter) {
        this.formatter = formatter;
//        throw new UnsupportedOperationException();
    }

    public void publishText(String text) {
        formatter.format(text);
//        throw new UnsupportedOperationException();
    }

}

File: exercises/week05/src/Strategy/Scala/TextEditor.scala
package Strategy.Scala

class TextEditor {

  var formatter: TextFormatter = _

  def this(textFormatter: TextFormatter) {
    this()
    this.formatter = textFormatter

//    throw new UnsupportedOperationException
  }


  def publishText(text: String) {

    formatter.format(text)
//    throw new UnsupportedOperationException
  }

}